### ISSUE TITLE
Cannot find created before payment method (paypal)

### ISSUE THREAD

**Issue by gvanto** [oc-shopaholic/oc-shopaholic-plugin#388]
I want to purchase Paypal plugin and looking at docs: https://octobercms.com/plugin/lovata-paypalshopaholic > Documentation

I can't seem to see where the 'created before payment method' is referred to here?:
![image](https://user-images.githubusercontent.com/8372918/130743228-83a75eee-4137-429e-a7c2-7a26b0179889.png)

Is there an example of setting up a Paypal payment in the docs somewhere? 
(no Paypal docs here: https://shopaholic.one/docs#/plugins/home)

Also, is there a way to create a Paypal subscription (instead of just a once-off order) ?

Thanks,
gvanto

**Comment by gvanto** [comment]
This is the payment method created - where do I put the (paypal) token?

![image](https://user-images.githubusercontent.com/8372918/132117017-9ef611d9-0224-4503-8673-b487b1edc95b.png)

**Comment by gvanto** [comment]
Anyone?

**Comment by kharanenka** [comment]
Hi! Did you install paypal composer packages via composer?

**Comment by gvanto** [comment]
Hello! Yes I did

**Comment by kharanenka** [comment]
Did you install v2 or v3 of omnipay packages?

**Comment by gvanto** [comment]
Looks like v3: from my composer.lock file:
![image](https://user-images.githubusercontent.com/8372918/134127622-f18da58a-54e0-4dd3-92a5-f9245cddd653.png)

**Comment by kharanenka** [comment]
Try to add this code in your custom plugin
```php
public function boot()
{
    $factory = Omnipay::getFactory();
    $factory->register('PayPal');
}
```

**Comment by gvanto** [comment]
Ok before doing this - is it possible to use Paypal Subscribe at all with checkout? I need a [paypal subscribe](https://developer.paypal.com/docs/paypal-payments-standard/integration-guide/subscribe-step-1/) to process payment for the product (not just a Paypal pay).

**Comment by kharanenka** [comment]
Unfortunately, I didn't have some case with using PayPal subscription. I think it is possible, but you should to do some minor improvments in your custom plugin.

**Comment by gvanto** [comment]
Surprised there's no case for it, since there's also a lovata shopaholic subscriptions plugin (which I also have), wouldn't Paypal subscriptions make sense to go with it?

**Comment by kharanenka** [comment]
You are right that plugins should work together out of the box. The plugins have been developed in different time. We will try to improve our products so that all plugins know how to work with each other.

**Comment by mennax** [comment]
Hi @kharanenka 
> Try to add this code in your custom plugin
> 
> ```
> public function boot()
> {
>     $factory = Omnipay::getFactory();
>     $factory->register('PayPal');
> }
> ```

this return an error when I choose PayPal from Gateway > Payment gateway:  
Class '\Omnipay\PayPal\Gateway' not found" on line 85 of /Users/mennax/Sites/vicinus2/vendor/omnipay/common/src/Common/GatewayFactory.php

but if I add this  

```
public function boot()
{
    $this->addEventListener();
    #$gateway = Omnipay::create('ExpressGateway');
    $factory = Omnipay::getFactory();
    $factory->register('PayPal');
    $factory->register('PayPal_Express');
}
```
And I choose PayPal_Express, then the 'Gateway' tab is now populated with a a lot of input fields.

Can you help me please?
